presto-main/src/main/java/com/facebook/presto/server/StatementResource.java

```diff
@@ -210,6 +213,19 @@ private static Response getQueryResults(Query query, Optional<Long> token, UriIn
         query.getResetSessionProperties().stream()
                 .forEach(name -> response.header(PRESTO_CLEAR_SESSION, name));
 
+        // add added prepare statements
+        query.getAddedPreparedStatements().entrySet().stream()
+                .forEach(entry -> {
+                    try {
+                        String encodedKey = encode(entry.getKey(), "UTF-8");
```

Comment:
Don't static import `encode` here since the name is generic and it's not clear that this is doing URL encoding (unfortunately, the method was not named `urlEncode`)